api/src/main/java/org/confeti/controllers/receiver/ReceiverController.java

```diff
@@ -1,25 +1,46 @@
 package org.confeti.controllers.receiver;
 
+import lombok.RequiredArgsConstructor;
 import org.confeti.dataobjects.Status;
-import org.confeti.dataobjects.receiver.Conference;
-import org.springframework.http.HttpStatus;
+import org.confeti.dataobjects.receiver.InputData;
+import org.confeti.service.ConferenceService;
+import org.confeti.service.ReportService;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
 import static org.confeti.controllers.Controllers.REST_API_PATH;
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
 @RestController
 @RequestMapping(value = REST_API_PATH + "/conference")
+@RequiredArgsConstructor
 public class ReceiverController {
 
+    private final ReportService reportService;
+    private final ConferenceService conferenceService;
+
     @PostMapping(consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
     @ResponseBody
-    public Mono<Status> handlePost(@RequestBody Conference conference) {
-        System.out.println(conference);
-        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED);
+    public Mono<ResponseEntity<Status>> handlePost(@RequestBody final InputData inputData) {
+        return Mono.justOrEmpty(inputData.getConference())
```

Comment:
You can do this pipeline in one line:
```
Mono.justOrEmpty(inputData.getConference())
         .switchIfEmpty(Mono.error(BAD_REQUEST_EXCEPTION))
         .map(InpuData::getReports)
         .flatMapMany(Flux::fromIterable)
         ....
         .onErrorReturn(ResponseEntity.badRequest().body(Status.fail()))
```

Reply:
Really? It doesn't work

Reply:
Sorry, I mean:
```
        return Mono.justOrEmpty(inputData.getConference())
                .switchIfEmpty(Mono.error(new Exception("bad request")))
                .doOnNext(conferenceService::upsert)
                .then(Mono.just(inputData))
                .map(data -> Objects.requireNonNullElse(data.getReports(), new ArrayList<Report>()))
                .flatMapMany(Flux::fromIterable)
                .map(report -> report.setConferences(Set.of(inputData.getConference())))
                .flatMap(reportService::upsert)
                .then()
                .thenReturn(ResponseEntity.ok(Status.ok()))
                .onErrorReturn(ResponseEntity.badRequest().body(Status.fail()));
```

but if we add default value of reports in InputData, we can replace `.then(Mono.just(inputData)).map(data -> Objects.requireNonNullElse(data.getReports(), new ArrayList<Report>()))` on `.then(Mono.just(inputData.getReports()))`